opencog/persist/sql/multi-driver/SQLAtomStorage.cc

```diff
@@ -93,14 +93,28 @@ class SQLAtomStorage::Response
 		LLConnection* _conn;
 
 	public:
-		Response(concurrent_stack<LLConnection*>& pool) : _pool(pool)
-		{
-			tname = "";
-			itype = 0;
-			intval = 0;
-			_conn = nullptr;
-			rs = nullptr;
-		}
+		Response(concurrent_stack<LLConnection*>& pool) :
+		    rs(nullptr),
+		    itype(0),
+		    name(nullptr),
+		    outlist(nullptr),
+		    height(0),
+		    floatval(0),
+		    stringval(nullptr),
+		    linkval(nullptr),
+		    _pool(pool),
+		    _conn(nullptr),
+		    table(nullptr),
+		    store(nullptr),
+		    pvec(nullptr),
+		    uvec(nullptr),
+		    tname(""),
+		    fltval(0),
+		    strval(nullptr),
+		    lnkval(nullptr),
+		    get_all_values(false),
+		    intval(0)
+		{}
```

Comment:
Is this a wise idea? Most of these values were not previously initialized, because I wanted to avoid the CPU overhead of actually initializing them. Most of these values are neither set, nor looked at; when they are looked at, they are also set. 

Perhaps the CPU overhead of initializing this is a drop in the bucket, and I shouldn't complain.  Perhaps it would be more efficient to just `bzero` the entire struct?

Reply:
I think with modern compiler you'll probably get something very efficient anyway, wiping a set of bytes is not that hard to optimize. Also, IIUC this code wraps around responses from the SQL DB so its whole execution time should be like nothing compared to SQL query/response time with networking or even interprocess communication.

But this improves reliability and correctness of the code, especially wrt to changes from random people from the Internet (like me) who prefer to think of constructed objects as something ready for the game in every aspect of it, so I think it's worth it.

Reply:
OK,, that's fine. lets keep this change then.